concent_api/api_testing_common.py

```diff
@@ -0,0 +1,187 @@
+import datetime
+import http.client
+import json
+import requests
+import sys
+
+from collections                    import namedtuple
+
+from golem_messages.exceptions      import MessageError
+from golem_messages.message         import ComputeTaskDef
+from golem_messages.message         import Message
+from golem_messages.message         import TaskToCompute
+from golem_messages.shortcuts       import dump
+from golem_messages.shortcuts       import load
+
+
+ProtocolConstants = namedtuple(
+    "ProtocolConstants",
+    [
+        "concent_messaging_time",
+        "subtask_verification_time",
+        "force_acceptance_time",
+        "token_expiration_time",
+    ]
+)
+
+
+def get_protocol_constants(cluster_url):
+    url                         = f"{cluster_url}/api/v1/protocol-constants/"
+    response                    = requests.get(url)
+    json                        = response.json()
+    concent_messaging_time      = json['concent_messaging_time']
+    subtask_verification_time   = json['subtask_verification_time']
+    force_acceptance_time       = json['force_acceptance_time']
+    token_expiration_time       = json['token_expiration_time']
+    constants                   = ProtocolConstants(
+            concent_messaging_time      = concent_messaging_time,
+            subtask_verification_time   = subtask_verification_time,
+            force_acceptance_time       = force_acceptance_time,
+            token_expiration_time       = token_expiration_time
+    )
```

Comment:
We talked about it earlier and there was something that you made me realize. If the server returns constants not listed in `ProtocolConstants`, this code won't notice. We want the code to crash in that case so that we can fix it. So please add a check that ensures that only the constants listed here are in the JSON response.

Alternatively you could use the form I showed in my previous comment - it's more concise and at the same time gives you that check:
``` python
constants = ProtocolConstants(**json)
```

By the way, we use 4 spaces per indent level. You have 6 here.

Reply:
done